File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/78-Subsets/78-Subsets.cpp
// Last updated: 8/4/2026, 7:23:21 PM
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        int n = nums.size();
        int subset = 1 << n;
        vector<vector<int>>result = {{}};

        for(int i = 1 ; i < subset ; i++){
            vector<int>sub_sets = {};
            for(int j = 0 ; j < n ; j++){
                if(i & (1 << j)){
                    sub_sets.push_back(nums[j]);
                }
            }
            result.push_back(sub_sets);
        }
        return result;
    }
};

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/78-Subsets/78-Subsets.java
// Last updated: 8/4/2026, 7:23:16 PM
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        int n=nums.length;
        List<Integer> ll=new ArrayList<>();
        List<List<Integer>> res=new ArrayList<>();
        solve(nums,n-1,ll,res);
        return res;
    }
    public void solve(int[] nums,int idx,List<Integer> ll,List<List<Integer>> res){
        if(idx < 0){
            res.add(new ArrayList<>(ll));
            return;
        }
        ll.add(nums[idx]);
        solve(nums,idx-1,ll,res);
        ll.remove(ll.size()-1);
        solve(nums,idx-1,ll,res);
    }
}